mycroft/skills/mycroft_skill/mycroft_skill.py

```diff
@@ -350,6 +350,8 @@ def converse(self, utterances, lang=None):
                                first entry is the user utt and the second
                                is normalized() version of the first utterance
             lang:       language the utterance is in, None for default
+            message:    a message object containing a message type with an
+                        optional JSON data packet
```

Comment:
I wonder if we should remove the `utterances` and `lang` arguments all together... Since the implementation doesn't support passing all three, having the docs / base class list them all.

This would make handling a message the default implementation while still allowing older skills using the utterances and lang signature using inspect.

This would make the utterance + lang signature deprecated in favor for the use of the message

Reply:
I think this makes sense if the idea is to move all converse interactions to use messages. Then a warning could be generated in skill_manager if a skill uses the old signature.